lsp-scala-plugin/src/main/scala/com/idiomaticsoft/lsp/scala/ScalaLSPPlugin.scala

```diff
@@ -54,7 +54,7 @@ class ScalaLSPPlugin extends AbstractUIPlugin {
 		wc.setAttribute(IJavaLaunchConfigurationConstants.ATTR_MAIN_TYPE_NAME, "scala.meta.metals.Main")
 		wc.setAttribute(IJavaLaunchConfigurationConstants.ATTR_DEFAULT_CLASSPATH, false)
 		wc.setAttribute(DebugPlugin.ATTR_CAPTURE_OUTPUT, false)
-		val vmParams = "-XX:+UseG1GC -XX:+UseStringDeduplication -Xss4m -Xms100m -Xmx2G"
+		val vmParams = "-XX:+UseG1GC -XX:+UseStringDeduplication -Xss4m -Xms100m -Xmx2G -Dmetals.http=true"
 		wc.setAttribute(IJavaLaunchConfigurationConstants.ATTR_VM_ARGUMENTS, wc.getAttribute(IJavaLaunchConfigurationConstants.ATTR_VM_ARGUMENTS, vmParams))
```

Comment:
I think you need to add a new line:

wc.setAttribute(IJavaLaunchConfigurationConstants.ATTR_PROGRAM_ARGUMENTS,  "-Dmetals.http=true")

The attribute is not for the VM but for the program

Reply:
Also, you can modify all these parameters to test in the Run Configuration dialogs for metals

Reply:
Tried that and it didn't seem to work when -D is put after the name of the Main class.

Reply:
ok, so these parameters are really properties that are set in the VM